### PR TITLE
[RLlib] Issue 21044: Improve error message for "multiagent" dict checks.

### PR DESCRIPTION
**Pull request by sven1977**
Issue 21044: Improve error message for "multiagent" dict checks.

## Why are these changes needed?

https://github.com/ray-project/ray/issues/21044

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/utils/multi_agent.py [hunk 1]
```diff
@@ -16,10 +16,27 @@ def check_multi_agent(config: PartialTrainerConfigDict) -> \
     Returns:
         Tuple consisting of the resulting (all fixed) multi-agent policy
         dict and bool indicating whether we have a multi-agent setup or not.
+
+    Raises:
+        KeyError: If `config` does not contain a "multiagent" key or if there
+        is an invalid key inside the "multiagent" config.
     """
+    if "multiagent" not in config:
+        raise KeyError(
```
Comment (avnishn) [inline rllib/utils/multi_agent.py:25]: Can we add a test that covers this?

File: rllib/utils/multi_agent.py [hunk 2]
```diff
@@ -16,10 +16,27 @@ def check_multi_agent(config: PartialTrainerConfigDict) -> \
...
+            "the 'multiagent' key defined!")
     multiagent_config = config["multiagent"]
+
     policies = multiagent_config.get("policies")
 
+    # Check for invalid sub-keys of multiagent config.
+    from ray.rllib.agents.trainer import COMMON_CONFIG
+    allowed = list(COMMON_CONFIG["multiagent"].keys())
+    if any(k not in allowed for k in multiagent_config.keys()):
+        raise KeyError(
```
Comment (avnishn) [inline rllib/utils/multi_agent.py:36]: Can we add a test that covers this?

File: rllib/execution/rollout_ops.py [hunk 3]
```diff
@@ -275,6 +275,14 @@ def __call__(self, samples: SampleBatchType) -> SampleBatchType:
         for policy_id in samples.policy_batches:
             batch = samples.policy_batches[policy_id]
             for field in self.fields:
+                if field not in batch:
+                    raise KeyError(
```
Comment (avnishn) [inline rllib/execution/rollout_ops.py:279]: Can we add a test that covers this?

**Review by avnishn — APPROVED** [review]
I think this is fine, pending some tests that show that the KeyErrors are being appropriately raised.

**Review by avnishn — DISMISSED** [review]

**Comment by sven1977** [comment]
Thanks @avnishn , great idea, will add those tests before merging.

**Review by avnishn — APPROVED** [review]

**Pull request merged**